### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Yehorf21**
[DEMO LINK](https://Yehorf21.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Comment by Yehorf21** [comment]
I added conversion of input to lower case, but the Demo Link didn't get updated yet

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,106 @@
...
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [isVisible, setIsVisible] = useState<Visibility>({
+    visible: false,
+    id: 0,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
```
Comment (IvanFesenko) [inline src/App.tsx:26]: You dont use await, so async is redundant
```suggestion
    function fetchData() {
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 2]
```diff
@@ -1,30 +1,88 @@
...
+import React, { useState } from 'react';
+import { Select } from '../../types/Select';
+
+interface Props {
+  onSelect: (val: Select) => void;
+  onInput: (value?: string) => void;
+}
+
+export const TodoFilter: React.FC<Props> = ({ onSelect, onInput }) => {
+  const [currentSelect, setCurrentSelect] = useState<Select>(Select.All);
```
Comment (IvanFesenko) [inline src/components/TodoFilter/TodoFilter.tsx:10]: I suggest move this state into App and use for get filtered todos, instead of filteredTodos state

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Looks good, Check this case. In case query not empty it should filter
https://github.com/mate-academy/react_dynamic-list-of-todos/assets/61904995/5b97f2ae-9c11-4598-b089-4c64ef7fe65a

**Pull request closed without merging**